acvm-repo/blackbox_solver/src/bigint.rs

```diff
@@ -84,8 +96,11 @@ impl BigIntSolver {
             }
             BlackBoxFunc::BigIntMul => lhs * rhs,
             BlackBoxFunc::BigIntDiv => {
+                if pedantic_solving && rhs == BigUint::ZERO {
+                    panic!("--pedantic-solving: BigIntDiv by zero");
```

Comment:
Should this be a `BlackBoxResolutionError` instead? Dividing by zero sounds like a user error.